# Housing Market

### Introduction:

This time we will create our own dataset with fictional numbers to describe a house market. As we are going to create random data don't try to reason of the numbers.

### Step 1. Import the necessary libraries

In [1]:
import numpy as np
import pandas as pd

### Step 2. Create 3 differents Series, each of length 100, as follows: 
1. The first a random number from 1 to 4 
2. The second a random number from 1 to 3
3. The third a random number from 10,000 to 30,000
---
[np.random.randint](https://numpy.org/doc/stable/reference/random/generated/numpy.random.randint.html)

In [16]:
s1 = pd.Series(np.random.randint(low=1, high=5, size=100, dtype='l'))
s2 = pd.Series(np.random.randint(low=1, high=4, size=100, dtype='l'))
s3 = pd.Series(np.random.randint(low=10000, high=30001, size=100, dtype='l'))

print(s1, s2, s3)

0     3
1     1
2     3
3     1
4     1
     ..
95    1
96    4
97    3
98    3
99    4
Length: 100, dtype: int32 0     1
1     2
2     1
3     3
4     1
     ..
95    2
96    1
97    3
98    3
99    2
Length: 100, dtype: int32 0     17668
1     24130
2     15985
3     15203
4     21236
      ...  
95    22219
96    23753
97    20473
98    11143
99    21148
Length: 100, dtype: int32


### Step 3. Let's create a DataFrame by joinning the Series by column

In [17]:
housemkt = pd.concat([s1, s2, s3], axis=1)
housemkt.head()

0  1      2
0  3  1  17668
1  1  2  24130
2  3  1  15985
3  1  3  15203
4  1  1  21236

### Step 4. Change the name of the columns to bedrs, bathrs, price_sqr_meter

In [18]:
# 내 풀이
# housemkt.columns = ['bedrs', 'bathrs', 'price_sqr_meter']

# 답지
housemkt.rename(columns={0:'bedrs', 1:'bathrs', 2:'price_sqr_meter'}, inplace=True)
housemkt.head()

bedrs  bathrs  price_sqr_meter
0      3       1            17668
1      1       2            24130
2      3       1            15985
3      1       3            15203
4      1       1            21236

### Step 5. Create a one column DataFrame with the values of the 3 Series and assign it to 'bigcolumn'
- Series를 DataFrame으로 바꿔주는 과정이 필요하다.

In [21]:
bigcolumn = pd.concat([s1, s2, s3], axis=0)
print(type(bigcolumn))

bigcolumn = bigcolumn.to_frame()
print(type(bigcolumn))

bigcolumn

<class 'pandas.core.series.Series'>
<class 'pandas.core.frame.DataFrame'>


0
0       3
1       1
2       3
3       1
4       1
..    ...
95  22219
96  23753
97  20473
98  11143
99  21148

[300 rows x 1 columns]

### Step 6. Oops, it seems it is going only until index 99. Is it true?

In [23]:
len(bigcolumn)

300

In [24]:
bigcolumn.reset_index()

index      0
0        0      3
1        1      1
2        2      3
3        3      1
4        4      1
..     ...    ...
295     95  22219
296     96  23753
297     97  20473
298     98  11143
299     99  21148

[300 rows x 2 columns]

### Step 7. Reindex the DataFrame so it goes from 0 to 299
- 기존의 인덱스를 drop하고 원래 데이터 형태도 바꾼..

In [25]:
bigcolumn.reset_index(drop=True, inplace=True)
bigcolumn

0
0        3
1        1
2        3
3        1
4        1
..     ...
295  22219
296  23753
297  20473
298  11143
299  21148

[300 rows x 1 columns]